docs/model/type/term.md

```diff
@@ -0,0 +1,77 @@
+# Terms
+
+Taxonomies (also referred to as vocabularies or ontologies) are used to
+organize and manage Terms used in various contexts throughout farmOS. These can
+be used for flagging, categorization, and organization of other record types.
+
+Most Terms in farmOS are user-defined, and the vocabularies are empty when
+farmOS is first installed. As [Logs](/model/type/log) and
+[Assets](/model/type/asset) are created, Terms that are used to describe them
+are automatically generated in the appropriate vocabularies. These can then be
+used to filter records in the future.
+
+Taxonomies are defined by modules, and are only available if their module is
+enabled. The modules included with farmOS define the following taxonomies:
+
+- Animal type
+- Log category
+- Material
+- Plant type
+- Season
+- Unit
+
+## ID
+
+Each Term will be assigned two unique IDs in the database: a universally unique
+identifier (UUID), and an internal numeric ID.
+
+The UUID will be unique **across** farmOS databases. The internal ID will only
+be unique to a **single** farmOS database. Therefore, the farmOS API uses UUIDs
+to ensure that IDs pulled from multiple farmOS databases do not conflict.
+Internally, farmOS modules use the internal IDs to perform CRUD operations.
+
+## Attributes
+
+Terms have a number of attributes that serve to describe their meta information.
+All Terms have the same standard set of attributes. Modules can add additional
+attributes.
+
+Attributes that are common to all Term types include:
+
+- Name
+- Description
+- Weight
+
+#### Name
+
+Terms must have a name.
+
+#### Description
+
+Optionally, Terms may have a description.
+
+#### Weight
+
+The Term's weight determines how it is ordered in the vocabulary. See related
+"Parent" relationship below for organizing Terms into a hierarchy.
+
+## Relationships
+
+All Terms have the same standard set of relationships. Modules can add
+additional relationships.
+
+### Standard relationships
+
+Relationships that are common to all Term types include:
+
+- Vocabulary
+- Parent
+
+#### Vocabulary
```

Comment:
Hmm yea I see your point. In this specific instance, though, if you open up a term in JSON:API, you will see "vocabulary" as one of the relationships. So it seems like we should keep that here, since this is a list of the relationships.

I agree that this "type" vs "vocabulary" piece in particular is an inconsistency, and does stem from Drupal's own internal legacy naming (the `taxonomy` module dates back to before entities/bundles entirely).

I did try to keep these lists of "Attributes" and "Relationships" identical to what one would see in the API, though. So this might just be a case where we compromise on that consistency a bit. 🤷

Reply:
Ah actually... looking more closely at the actual JSON for logs and terms I realize logs also have `log_type` as a relationship, and it is `vid` in the term. But, we don't include `log_type` in the "relationships" overview in `log.md`.

So maybe we should just omit `vocabulary` form `term.md` as well? That would make them consistent. But it would mean we are intentionally omitting those relationships. But like you said, maybe that's more of a Drupalism than a farmOS data model piece...

Reply:
Here is a commit that does this: https://github.com/farmOS/farmOS/pull/415/commits/10c1da4883a1b9266257593b9d26821b7fa2e46d - think that makes sense?

Reply:
If you send a term to the server with its `type` field set, but no `vid` under `relationships` it will accept it, right? But if the `type` is missing, it will reject it, even if the `vid` is there? I'm just assuming, haven't tried, but for me that would clinch it. `vid` and "vocabulary" as a whole is a detail of the Drupal implementation, which may be necessary for Drupal, but I don't see any reason why it would be necessary in other implementations.

Reply:
I believe that's true, yes.

Reply:
The `log_type` and `vid` relationships are sort of like "computed" relationships... is my understanding. They provide quick access to the actual resource that is referenced by the `type` string. So it's a helper, but not really "core data model" IMO. I think we can omit them.

Reply:
This reminds me of the issue of "computed" values, and how to distinguish them... :thinking: